db_stress_tool/db_stress_test_base.cc

```diff
@@ -1682,6 +1699,9 @@ Status StressTest::TestCheckpoint(ThreadState* thread,
   if (s.ok()) {
     Options options(options_);
     options.listeners.clear();
+    db_stress_listener_ = new DBStressCompactionListener();
+    options.listeners.emplace_back(db_stress_listener_);
```

Comment:
This looks wrong to me.  The listeners stores a std::vector<std::shared_ptr<EventListener>>.  You are also storing the same pointer outside of the vector (in db_stress_listener_).  This seems like it can/will lead to a double delete (if you clear the vector and then delete it) or potentially a problem of accessing the memory after its been deleted.